### PR TITLE
DOM-35162 - Remove launcher pod, rename mpijob to mpicluster

### PR DESCRIPTION
**Pull request by adp312**
I see now that the rename has messed everything up, as it affected many files. I should have done it in a separate commit... sorry...

The essential changes are in `pkg/cluster/mpi` directory. Specifically, `statusupdate.go` contains a new logic of tracking status of the cluster.

### CODE REVIEW

File: pkg/cluster/mpi/statusupdate.go [hunk 1]
```diff
@@ -105,3 +98,20 @@ func (c statusUpdateComponent) Reconcile(ctx *core.Context) (ctrl.Result, error)
...
+	objKey := client.ObjectKey{
+		Name:      serviceName(cr),
+		Namespace: cr.Namespace,
+	}
+
+	if err := ctx.Client.Get(ctx, objKey, &ep); err != nil && !apierrors.IsNotFound(err) {
+		return 0, fmt.Errorf("cannot fetch endpoints: %w", err)
+	}
+
+	if len(ep.Subsets) == 1 {
```
Comment (ddl-kevin) [inline pkg/cluster/mpi/statusupdate.go:113]: This seems a little brittle, I'm not sure how this works.

File: pkg/cluster/mpi/statusupdate.go [hunk 2]
```diff
@@ -105,3 +98,20 @@ func (c statusUpdateComponent) Reconcile(ctx *core.Context) (ctrl.Result, error)
...
 }
+
+func getActivePodCnt(ctx *core.Context, cr *dcv1alpha1.MPICluster) (int, error) {
+	var ep corev1.Endpoints
+	objKey := client.ObjectKey{
+		Name:      serviceName(cr),
+		Namespace: cr.Namespace,
+	}
+
+	if err := ctx.Client.Get(ctx, objKey, &ep); err != nil && !apierrors.IsNotFound(err) {
```
Comment (ddl-kevin) [inline pkg/cluster/mpi/statusupdate.go:109]: Do you know if this makes an API call? Ideally this is only querying a local cache.

**Review by ddl-kevin — APPROVED** [review]
LGTM, I have some questions on how we determine activePodCnt.

File: pkg/cluster/mpi/statusupdate.go (hunk 2 above)
Comment (adp312) [reply pkg/cluster/mpi/statusupdate.go:109]: This is really a copy-paste of Sonny's code from the portion that I have removed. Agree that this logic seems somewhat fragile. I will look at what's going on behind the scene here.

**Pull request merged**